_posts/2015-06-02-anatomy-of-service-objects-in-rails.md

```diff
@@ -0,0 +1,333 @@
+---
+title: Anatomy of a Rails Service Object
+layout: posts
+author: Dave Copeland
+author_url: 'http://naildrivin5.com'
+published: true
+location: "Washington, DC"
+---
+
+We've given up on “fat models, skinny controllers” as a design style for our Rails apps—in fact we abandoned it before we started.  Instead, we factor our code into special-purpose classes, commonly called _service objects_.  We've thrashed on exactly how these classes should be written, so this post is going to outline what I think is the most successful way to create a service object.
+
+## Purpose of a Service Object
+
+A service object's job is to hold the code for a particular bit of business logic.  For example, when we process a customer's returned items in the warehouse, we use an instance of `ReturnProcessor` to handle that.  Unlike the “fat model” style, where a small number of objects contain many, many methods for all necessary logic, using service objects results in many classes, each of which are single purpose.
```

Comment:
Do you think it's worth a link to an article on the [SRP](http://en.wikipedia.org/wiki/Single_responsibility_principle) for those that may not be familiar with it?